lib/symboldatabase.cpp

```diff
@@ -3703,8 +3701,10 @@ bool Variable::arrayDimensions(const Settings& settings, bool& isContainer)
     isContainer = false;
     const Library::Container* container = (mTypeStartToken && mTypeStartToken->isCpp()) ? settings.library.detectContainer(mTypeStartToken) : nullptr;
     if (container && container->arrayLike_indexOp && container->size_templateArgNo > 0) {
-        const Token* tok = Token::findsimplematch(mTypeStartToken, "<");
```

Comment:
well in theory if there is no "<" as expected in the statement then this could find a "<" in another statement.
It does not seem that Library::detectContainer enforce that there must be a "<".

Reply:
Some containers dont have angle brackets like `std::string` or `std::string_view`.

Reply:
yes :+1:

Reply:
but it feels like `container->size_templateArgNo` should be 0 for those